src/components/burger-constructor/burger-constructor.tsx

```diff
@@ -1,44 +1,116 @@
-import React, { useEffect, useState } from "react";
+import React, { useContext, useEffect, useReducer, useState } from "react";
 import {ConstructorElement, DragIcon, Button, CurrencyIcon}  from '@ya.praktikum/react-developer-burger-ui-components';
 import styles from './burger-constructor.module.css';
 import Modal from '../modal/modal';
 import OrderDetails from './../order-details/order-details';
 import PropTypes from 'prop-types';
 import ingredientItemPropTypes from '../../utils/ingredient-item-prop-types';
+import IngridientsDataContext from "../services/IngridientsDataContext";
+import { Ingredient } from "../model/ingredient";
+import { cartData } from "../services/cartContext";
+import ConstructorDataContext from "../services/constructorDataContext";
+
+const sentOrderUrl = 'https://norma.nomoreparties.space/api/orders';
 
 function BurgerConstructor(props: any){
 
-  const [data, setData] = useState([] as any[]);
+  const ingredientData = useContext(IngridientsDataContext) as Ingredient[];
   const [orderOpen, setOrderOpen] = useState(false);
 
-  useEffect(() => {
-    setData(props.data);
-  }, [props.data, orderOpen])
+  const [orderData, setOrderData] = useState(null);
+  const [cartState, dispatchCart] = useContext(ConstructorDataContext);
 
+  // Открыть окно заказа
   const openOrder = () => {
-    setOrderOpen(true);
+    let ingredients: any[] = [];
+    cartState.items.forEach((item: Ingredient) => {
+      ingredients.push(item._id);
+    });
```

Comment:
Здесь тоже, так как нужны только `id` ингредиентов, то можно воспользоваться методом `map`
```suggestion
    const ingredientIds: string[] = cartState.items.map(( {_id} : Ingredient) => _id);
```
С помощью этого метода можно проходить по каждому элементу массива и видоизменять его как нужно